src/v/kafka/server/replicated_partition.h

```diff
@@ -61,6 +68,13 @@ class replicated_partition final : public kafka::partition_proxy::impl {
 
     ss::future<std::vector<cluster::rm_stm::tx_range>>
     aborted_transactions(model::offset base, model::offset last) final {
+        model::offset local_kafka_start_offset = _translator->from_log_offset(
+          _partition->start_offset());
+        if (base < local_kafka_start_offset) {
+            // TODO: is this correct?
+            co_return std::vector<cluster::rm_stm::tx_range>{};
```

Comment:
@rystsov can you comment if it is okay to return empty range for offsets less than the local start offset here?

Reply:
No, it doesn't look right. When Redpanda handles a fetch request it reads a range of records [there](https://github.com/vectorizedio/redpanda/blob/0dd630b3e1517b32142bb32f89f77c8505ed0a89/src/v/kafka/server/handlers/fetch.cc#L153) and then it uses `aborted_transactions` to find aborted transactions which intersect with the read offset range. When `local_kafka_start_offset < last` there still may be an intersection.

Reply:
Even if we restrict the interval to local offsets, it still leaves the question open about what to do when we read from remote segments - presumably we still need the info about aborted transactions for them, but I don't know if `rm_stm` still contains this information and in fact if we can get this information anywhere else. We need to discuss this further.